atest/testdata/standard_libraries/process/terminate_process.robot

```diff
@@ -98,8 +98,9 @@ Terminate all processes
 Terminating all empties cache
     Some process
     Terminate All Processes    kill=True
-    ${handle} =    Some Process
-    Should Be Equal    ${handle}    ${1}
+    ${process_object}=    Some Process
+    ${index}=    Get Process Object Index     ${process_object}
+    Should Be Equal    ${index}    ${1}
```

Comment:
I don't think it's a good idea to add separate `Get Process Object Index` keyword for just this usage. Other alternatives to make sure the cache is empty:
1. Use `Switch Process    1` and handle resulting error by adding `[Documentation]` that specifies what error should occur. Although `Start Process` now returns Popen objects, `Switch Process` should work with indices. But not when there are no processes.
2. Use `BuiltIn.Get Library Instance` to get access to the library itself and check that `lib._processes` is empty. Accessing an attribute starting with `_` is a bit questionable but ought to be fine in tests.